solver/src/liquidity/offchain_orderbook.rs

```diff
@@ -175,61 +138,24 @@ pub mod tests {
         );
     }
 
-    #[test]
-    fn executed_buy_amount_returns_err_on_overflows() {
-        let order = Order {
-            order_creation: OrderCreation {
-                kind: OrderKind::Sell,
-                ..Default::default()
-            },
-            ..Default::default()
-        };
-
-        // mul
-        let executed_amount = U256::MAX;
-        let sell_price = U256::from(2);
-        let buy_price = U256::one();
-        assert!(executed_buy_amount(
-            &order,
-            executed_amount,
-            Price {
-                sell_price,
-                buy_price
-            }
-        )
-        .is_none());
-
-        // div
-        let executed_amount = U256::one();
-        let sell_price = U256::one();
-        let buy_price = U256::zero();
-        assert!(executed_buy_amount(
-            &order,
-            executed_amount,
-            Price {
-                sell_price,
-                buy_price
-            }
-        )
-        .is_none());
-    }
-
     #[test]
     fn adds_unwrap_interaction_for_sell_order_with_eth_flag() {
         let native_token_address = H160([0x42; 20]);
         let sell_token = H160([0x21; 20]);
         let native_token = dummy_contract!(WETH9, native_token_address);
 
         let executed_amount = U256::from(1337);
+        let executed_buy_amount = U256::from(2 * 1337);
+
         let prices = hashmap! {
             native_token.address() => U256::from(100),
             sell_token => U256::from(200),
         };
-        let executed_buy_amount = U256::from(2 * 1337);
         let order = Order {
             order_creation: OrderCreation {
                 buy_token: BUY_ETH_ADDRESS,
                 sell_token,
+                sell_amount: 1337.into(),
```

Comment:
I see that we insert sell amounts here where there were none before. Given that they were previously defaults of zero, how does this relate to the PRs intended purpose?

Reply:
Its because adding trades to the settlement encode now validates some values required for computing executed buy, sell and fee amount. Having them as 0 makes orders not tradable.